src/osd/PGLog.cc

```diff
@@ -938,8 +938,7 @@ void PGLog::_write_log_and_missing(
       if (!missing.is_missing(obj, &item)) {
 	to_remove.insert(key);
       } else {
-	uint64_t features = missing.may_include_deletes ? CEPH_FEATURE_OSD_RECOVERY_DELETES : 0;
-	encode(make_pair(obj, item), (*km)[key], features);
+	encode(make_pair(obj, item), (*km)[key], CEPH_FEATUREMASK_SERVER_OCTOPUS);
```

Comment:
This is what I meant! I will also run this through octopus upgrade tests to make sure everything looks good.

Reply:
https://pulpito.ceph.com/teuthology-2021-09-17_14:34:25-upgrade:octopus-x-master-distro-basic-smithi/ - no related failures